### PR TITLE
[LPT] Introduce new granularity attribute instead of OperationPerTensorQuantizationRestriction

### PR DESCRIPTION
**Pull request by eshoguli**
### Details:
 - *Introduce new granularity attribute instead of OperationPerTensorQuantizationRestriction*

### Tickets:
 - *73725*

ARM PR: https://github.com/openvinotoolkit/openvino_contrib/pull/362
product-config PR: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/1055

### CODE REVIEW

**Comment by eshoguli** [comment]
@vzinovie, @kblaszczak-intel, please review

**Comment by eshoguli** [comment]
@AlexKoff88, @nkogteva, just for your info

File: docs/snippets/lpt_intel_cpu_plugin.cpp [hunk 1]
```diff
@@ -70,29 +70,29 @@ manager.run_passes(nGraphFunc);
...
             {1, {ngraph::element::i8}},
         }),
     });
 
     // Low precision transformations plugin specific configuration: per-tensor quantization operations definition
-    auto perTensorQuantization = std::vector<OperationPerTensorQuantizationRestriction>({
-        OperationPerTensorQuantizationRestriction::create<ngraph::opset1::Convolution>({0}),
-        OperationPerTensorQuantizationRestriction::create<ngraph::opset1::ConvolutionBackpropData>({0})
+    auto perTensorQuantization = std::vector<QuantizationGranularityRestriction>({
+        QuantizationGranularityRestriction::create<ngraph::opset1::Convolution>({0}),
```
Comment (AlexKoff88) [inline docs/snippets/lpt_intel_cpu_plugin.cpp:94]: Since it is anyway quantization related stuff you can call it just `GranularityRestriction`.

**Review by vzinovie — APPROVED** [review]
LGTM, discussed offline. Approved if validation result will be good.

**Comment by azhogov** [comment]
CI passed: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/1055

**Pull request merged**